src/ewatercycle/_forcings/caravan.py

```diff
@@ -0,0 +1,270 @@
+        from ewatercycle.forcing import sources
+
+        path = Path.cwd()
+        forcing_path = path / "Forcing"
+        experiment_start_date = "1997-08-01T00:00:00Z"
+        experiment_end_date = "2005-09-01T00:00:00Z"
+        HRU_id = 1022500
+
+        camels_forcing = sources['LumpedCaravanForcing'].retrieve(
+                                    start_time = experiment_start_date,
+                                    end_time = experiment_end_date,
+                                    directory = forcing_path / "Camels",
+                                    basin_id = f"camels_0{HRU_id}"
+                                                                )
+        which gives somthing like:
+
+    .. code-block:: python
+
+        LumpedCaravanForcing(
+        start_time='1997-08-01T00:00:00Z',
+        end_time='2005-09-01T00:00:00Z',
+        directory=PosixPath('/home/davidhaasnoot/eWaterCycle-WSL-WIP/Forcing/Camels'),
+        shape=PosixPath('/home/davidhaasnoot/eWaterCycle-WSL-WIP/Forcing/Camels/shapefiles/camels_01022500.shp'),
+        filenames={
+            'tasmax':
+             'camels_01022500_1997-08-01T00:00:00Z_2005-09-01T00:00:00Z_tasmax.nc',
+            'tasmin':
+            'camels_01022500_1997-08-01T00:00:00Z_2005-09-01T00:00:00Z_tasmin.nc',
+            'evspsblpot':
+            'camels_01022500_1997-08-01T00:00:00Z_2005-09-01T00:00:00Z_evspsblpot.nc',
+            'pr': 'camels_01022500_1997-08-01T00:00:00Z_2005-09-01T00:00:00Z_pr.nc',
+            'tas': 'camels_01022500_1997-08-01T00:00:00Z_2005-09-01T00:00:00Z_tas.nc',
+            'Q': 'camels_01022500_1997-08-01T00:00:00Z_2005-09-01T00:00:00Z_Q.nc'
+        }
+        )
+
+
+    More in depth notebook van be found here:
+    https://gist.github.com/Daafip/ac1b030eb5563a76f4d02175f2716fd7
+    """
+
+    @classmethod
+    def generate(  # type: ignore[override]
+        cls: Type["CaravanForcing"],
+        start_time: str,
+        end_time: str,
+        directory: str,
+        variables: tuple[str, ...] = (),
+        shape: str | Path | None = None,
+        dataset: str | Dataset | dict = "unused",
+        **kwargs,
+    ) -> "CaravanForcing":
+        """Retrieve caravan for a model.
+
+        Args:
+            start_time: Start time of forcing in UTC and ISO format string e.g.
+                'YYYY-MM-DDTHH:MM:SSZ'.
+            end_time: nd time of forcing in UTC and ISO format string e.g.
+                'YYYY-MM-DDTHH:MM:SSZ'.
+            directory: Directory in which forcing should be written.
+                If not given will create timestamped directory.
+            variables: Variables which are needed for model,
+                if not specified will default to all.
+            shape: (Optional) Path to a shape file.
+                If none is specified, will be downloaded automatically.
+            dataset: Unused
+
+            **kwargs:
+                basin_id: str containing the wanted basin_id.
```

Comment:
Can you add where the user can lookup the basin ID?

Reply:
See this is where geopandas would be nice :p, but will add a function which lets the user explore the dataset(ids).

Reply:
I was hoping more for something like a webpage listing the basin_id's together with river/basin names, country, etc. I guess that doesn't exist?

Reply:
Nope, but what i do now is list the datasets in a seperate function

Reply:
For advanced users this will do, for more novice users choosing a single catchment: likely they will have to download the combinedshapefile, load this into GIS and pick one

Reply:
> I was hoping more for something like a webpage listing the basin_id's together with river/basin names, country, etc. I guess that doesn't exist?

would be pretty easy to make an interactive folium/leaflet map out of this actually! Similar to [this](https://nbviewer.org/github/Daafip/KNMI-Folium-Map/blob/main/Hourly%20%26%20daily%20station%20data%20KNMI%202023-03-21.html)

Reply:
Lets continue this discussion in issues: #398